jena-arq/src/main/java/org/apache/jena/sparql/engine/join/JoinKey.java

```diff
@@ -17,96 +17,125 @@
  */
 
 package org.apache.jena.sparql.engine.join;
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
 
 import org.apache.jena.sparql.core.Var ;
 
 /** JoinKey for hash joins */
-public final class JoinKey implements Iterable<Var>
+public final class JoinKey extends ImmutableUniqueList<Var>
```

Comment:
Unsure whether JoinKey should extend ImmutableUniqueList or include it as a field. ImmutableUniqueList provides an indexed List.indexOf() method - it could be seen as an internal class.

Reply:
I would have thought `ImmutableUniqueList` was an internal class.